### ISSUE TITLE
System.Text.Json tests crashing outerloop on Unix

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#39081]
e.g.
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38804~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190701.1/workItem/System.Text.Json.Tests/wilogs

cc: @ahsonkhan, @steveharter, @bartonjs

**Comment by bartonjs** [comment]
Those are all `SIGKILL`.  Looks like something sent the process that signal after 20 seconds (25 for the Alpine one).

**Comment by steveharter** [comment]
Have we see this in the past? When does it occur -  Debug.Assert? Machine rebooting, infrastructure issues?

i.e. is this a potential issue with S.T.Json?

**Comment by steveharter** [comment]
@stephentoub these are innerloop failures (not outerloop), correct?

**Comment by steveharter** [comment]
FYI [one ](https://github.com/dotnet/corefx/blob/359762251831ef498814f6718a54f5b91439d974/src/System.Text.Json/tests/JsonEncodedTextTests.cs#L285) of the tests allocate a 800MB contiguous block (and another 400MB) which seems excessive and could fail depending on environment.

**Comment by stephentoub** [comment]
> these are innerloop failures (not outerloop), correct?

@steveharter, actually this was from an outerloop CI run.

**Comment by steveharter** [comment]
> these are innerloop failures (not outerloop), correct?
>> @steveharter, actually this was from an outerloop CI run.

Yes I noticed that right after I added the comment, then deleted the comment, now it came back :(.

**Comment by steveharter** [comment]
Suggest moving this to future and re-open if this appears to be a real product bug.

My best guess is an out of memory test issue.

**Comment by stephentoub** [comment]
Another:
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F39311~2Fmerge/test~2Ffunctional~2Fcli~2Fouterloop~2F/20190710.6/workItem/System.Text.Json.Tests/wilogs
```
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 12 of 1085 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 2)
/root/helix/work/workitem
./RunTests.sh: line 161:    16 Killed                  "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
----- end Thu Jul 11 02:03:38 UTC 2019 ----- exit code 137 ----------------------------------------------------------
exit code 137 means SIGKILL Killed eg by kill
```

**Comment by stephentoub** [comment]
Another three:
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F39352~2Fmerge/test~2Ffunctional~2Fcli~2Fouterloop~2F/20190710.5/workItem/System.Text.Json.Tests/wilogs
```
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 12 of 1085 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 2)
./RunTests.sh: line 161:  2860 Killed                  "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
~/work/9d01a7a2-3b33-4c49-873c-728e2d9ca46b/Work/23dbdc49-d906-4557-a040-069e56b3f862/Exec
----- end Thu Jul 11 02:23:09 UTC 2019 ----- exit code 137 ----------------------------------------------------------
```
and
```
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 12 of 1085 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 2)
./RunTests.sh: line 161: 23678 Killed                  "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/datadisks/disk1/work/dd3b7795-54f5-48a1-bfe3-e35b23399652/Work/7b42349e-043f-40d3-8884-e17caf24f9c1/Exec
----- end Thu Jul 11 01:58:39 UTC 2019 ----- exit code 137 ---------------------------------------------------------
```
and
```
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 12 of 1085 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 2)
./RunTests.sh: line 161: 49771 Killed                  "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
~/work/9416c89d-89b4-493b-83e7-64a2176a6ff0/Work/f91db62f-1eee-4edf-aaa4-b0f44035fa46/Exec
----- end Thu Jul 11 01:52:42 UTC 2019 ----- exit code 137 ----------------------------------------------------------
```

**Comment by stephentoub** [comment]
Continues to take out outerloop runs, e.g.
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F39474~2Fmerge/test~2Ffunctional~2Fcli~2Fouterloop~2F/20190715.4/workItem/System.Text.Json.Tests/wilogs
```
Alpine.39.Amd64.Open-x64:Debug
Details from Job ef328e92-5763-4ff4-a494-657bca52fb11
ExitCode: 137
Ran on Machine: b000FRS
Get Repro environment
Console 
Executed on b000FRS using docker image mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.9-helix-09ca40b-20190620160617
running $HELIX_CORRELATION_PAYLOAD/scripts/0e4d5134e4244e3c86baf21af45c4458/execute.sh in /datadisks/disk1/work/ef328e92-5763-4ff4-a494-657bca52fb11/Work/e3faad63-ea49-4359-9ce5-77ba4b77f1da/Exec max 900 seconds

Output:
[BEGIN EXECUTION]
+ ./RunTests.sh --runtime-path /root/helix/work/correlation
----- start Mon Jul 15 16:01:54 UTC 2019 =============== To repro directly: =====================================================
pushd .
/root/helix/work/correlation/dotnet exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/root/helix/work/workitem /root/helix/work/workitem
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 12 of 1129 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 2)
/root/helix/work/workitem
./RunTests.sh: line 161:    17 Killed                  "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -trait category=OuterLoop -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
----- end Mon Jul 15 16:02:19 UTC 2019 ----- exit code 137 ----------------------------------------------------------
exit code 137 means SIGKILL Killed eg by kill
```

**Comment by steveharter** [comment]
We need to get a dump or call stack on the next hit.

**Comment by stephentoub** [comment]
> We need to get a dump or call stack on the next hit.

This one just occurred... do we have a dump from it?

If not, what are we going to do differently to enable getting a dump from the next one?

**Comment by steveharter** [comment]
I will attempt to repro this locally on one of the platforms that failed (Alpine, Ubuntu).

The repro tool wasn't working as of a few hours ago and now history seems to be down (i.e. the links above don't show any information).

@ViktorHofer any notes or thoughts on getting a repro? Is the repro tool supposed to be working? Sometimes the CI machines appear capture the crash dumps but not lately?

**Comment by karelz** [comment]
Fixed in master (5.0) in PR #39751 and in release/3.0 in PR #39784.
